### PR TITLE
feat(juju): add `jclean` function to destroy all controllers, and other aliases

### PR DESCRIPTION
**Pull request by sed-i**
## Standards checklist:
- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:
- add `jclean` function to destroy all controllers
- use the `jsh` prefix for `juju show-*` commands
- drop `--storage` from plain `status`
- introduce additional aliases

## Other comments:
NTA

### CODE REVIEW

**Comment by jsoref** [comment]
Hi, you're going to need to rebase this PR onto master in order to pass the spell checker which is going to ask you to add three lines to `.github/actions/spelling/expect.txt`:

```
jclean
jctl
unregister
```

(You can insert them alphabetically, or just place them somewhere.)

**Comment by sed-i** [comment]
@jsoref after rebasing the CI action pass without adding anything new to the spelling `expect.txt`.

File: plugins/juju/juju.plugin.zsh [hunk 1]
```diff
@@ -95,6 +95,18 @@ jaddr() {
...
+# Destroy all controllers
+jclean() {
+  if (( ! ${+commands[jq]} )); then
+    echo "jq is required but could not be found." >&2
+    return 1
+  fi
+  
+  juju controllers --format=json | jq '.controllers | keys[]' | timeout 2m xargs -I {} juju destroy-controller --destroy-all-models --destroy-storage --force --no-wait -y {}
+  juju controllers --format=json | jq '.controllers | keys[]' | timeout 2m xargs -I {} juju kill-controller -y -t 0 {}
+  juju controllers --format=json | jq '.controllers | keys[]' | timeout 10s xargs -I {} juju unregister {}
```
Comment (mcornella) [inline plugins/juju/juju.plugin.zsh:107]: Can we maybe cache these instead of calling 3 times?

File: plugins/juju/README.md [hunk 2]
```diff
@@ -38,6 +38,7 @@ Naming convention:
 
 | Alias    | Command                                                                               | Description                                                       |
 |----------|---------------------------------------------------------------------------------------|-------------------------------------------------------------------|
+| `jctl`   | `juju controllers`                                                                    | List all controllers                                              |
```
Comment (mcornella) [inline plugins/juju/README.md:41]: This alias is not actually added in the code.

File: plugins/juju/juju.plugin.zsh (hunk 1 above)
Comment (mcornella) [reply plugins/juju/juju.plugin.zsh:107]: Also this function is super destructive. Maybe adding a confirmation prompt before is a good idea.

**Comment by mcornella** [comment]
Reopen if you still want to work on this. I still feel though that `jclean` is superdestructive.

**Comment by sed-i** [comment]
Sorry about the radio silence.
I agree.
Will patch and reopen soon.
Thanks!

**Comment by mcornella** [comment]
Sounds good!

File: plugins/juju/README.md (hunk 2 above)
Comment (sed-i) [reply plugins/juju/README.md:50]: Added

File: plugins/juju/juju.plugin.zsh (hunk 1 above)
Comment (sed-i) [reply plugins/juju/juju.plugin.zsh]: Done.

**Comment by sed-i** [comment]
@mcornella Could you please take a look?

**Comment by mcornella** [comment]
This no longer does just one thing. You should rebase and recreate the commits so that each change is one item in the changelog, and properly announce breaking changes (such as renaming an alias) using the [Commit formatting guidelines](https://github.com/ohmyzsh/ohmyzsh/blob/master/CONTRIBUTING.md#commit-guidelines).

**Comment by mcornella** [comment]
For tips on how to do an interactive rebase see https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d.

**Comment by sed-i** [comment]
> This no longer does just one thing. You should rebase and recreate the commits so that each change is one item in the changelog, and properly announce breaking changes (such as renaming an alias) using the [Commit formatting guidelines](https://github.com/ohmyzsh/ohmyzsh/blob/master/CONTRIBUTING.md#commit-guidelines).

Done

File: plugins/juju/juju.plugin.zsh [hunk 3]
```diff
@@ -123,5 +165,6 @@ jreld() {
 wjst() {
   local interval="${1:-5}"
   shift $(( $# > 0 ))
-  watch -n "$interval" --color juju status --relations --storage --color "$@"
+  watch -n "$interval" --color juju status --relations --color "$@"
```
Comment (mcornella) [inline plugins/juju/juju.plugin.zsh:168]: Is this one required? It seems like a nitpick that isn't worth a breaking change commit.

**Pull request closed without merging**